libs/ledgerjs/packages/keyring-eth/src/mapper/TxMapper.ts

```diff
@@ -0,0 +1,79 @@
+import { getSerializedTransaction } from "@ledgerhq/coin-evm/transaction";
+import { TransactionSubset } from "@ledgerhq/context-module";
+import { SupportedTransaction } from "../model/Transaction";
+import LL from "@ledgerhq/coin-evm/types/index";
+import { ethers } from "ethers";
+import BigNumber from "bignumber.js";
+
+type MapTransactionResult = {
+  transactionRaw: string;
+  transactionSubset: TransactionSubset;
+};
+
+export class TxMapper {
+  mapTransaction(tx: SupportedTransaction): MapTransactionResult {
```

Comment:
Couldn't you simplify that code and get rid of isXXXTransaction functions using generics and `instanceof`?
something like:

    mapTransaction<T extends LL.Transaction | ethers.Transaction>(tx: T) {
        if (tx instanceof LL.Transaction) {
            this.mapLLTx(tx);
        } else {
            this.mapEthersv5Tx(tx);
        }
    }

Reply:
Unfortunately, `instanceof` works only with classes, not types

Reply:
(generic is maybe not even necessary, it can probably work the same with the SupportedTransaction you defined?

Reply:
:crying_cat_face: